static/js/lib/learning_resources.js

```diff
@@ -27,24 +34,131 @@ export const availabilityLabel = (availability: ?string) => {
   }
 }
 
+export const parseDateFilter = (filter: string) => {
+  const format = /(now)(\+)?(\d+)?([Md])?/
+  const match = format.exec(filter)
+  if (match) {
+    let dt = moment()
+    if (match[3] && match[4]) {
+      dt = dt.add(match[3], match[4] === "d" ? "days" : "months")
+    }
+    return dt
+  }
+}
+
+export const inDateRanges = (run: CourseRun, availabilities: Array<string>) => {
+  let inRange = false
+  availabilities.forEach(availability => {
+    if (AVAILABILITY_MAPPING[availability]) {
+      const from = parseDateFilter(
+        AVAILABILITY_MAPPING[availability].filter.from
+      )
+      const to = parseDateFilter(AVAILABILITY_MAPPING[availability].filter.to)
+      const startDate = run.best_start_date
+        ? moment(run.best_start_date, dateFormat)
+        : null
+      if (
+        ((isNil(startDate) && availability === AVAILABLE_NOW) ||
+          isNil(from) ||
+          // $FlowFixMe: okay to compare moments
+          startDate >= from) &&
+        ((isNil(startDate) && availability === AVAILABLE_NOW) ||
+          isNil(to) ||
+          // $FlowFixMe: okay to compare moments
+          startDate <= to)
+      ) {
+        inRange = true
+      }
+    }
+  })
+  return inRange
+}
+
+export const bestRunLabel = (run: ?CourseRun) => {
+  if (!run) {
+    return AVAILABILITY_MAPPING[AVAILABLE_NOW].label
+  }
+  for (const range in AVAILABILITY_MAPPING) {
+    if (inDateRanges(run, [range])) {
+      return AVAILABILITY_MAPPING[range].label
+    }
+  }
+}
+
+export const compareRuns = (firstRun: CourseRun, secondRun: CourseRun) =>
+  moment(firstRun.best_start_date || defaultStartDate, dateFormat).diff(
+    moment(secondRun.best_start_date || defaultStartDate, dateFormat),
+    "hours"
+  )
+
+export const bestRun = (runs: Array<CourseRun>) => {
+  const now = moment()
+
+  // Runs that are running right now
+  const currentRuns = runs.filter(
+    run =>
+      moment(run.best_start_date || defaultStartDate, dateFormat).diff(now) <=
+        0 &&
```

Comment:
I think it might be a little clearer for these bits of `.diff` logic to use moments `.isBefore` and `.isAfter` helper methods - you can do `momentObj.isBefore()` and if no argument is passed it defaults to `moment.now()`

Reply:
✔️